modules/aggs-matrix-stats/src/main/java/org/elasticsearch/search/aggregations/matrix/stats/InternalMatrixStats.java

```diff
@@ -157,7 +157,7 @@ MatrixStatsResults getResults() {
 
     @Override
     public XContentBuilder doXContentBody(XContentBuilder builder, Params params) throws IOException {
-        if (results != null && results.getFieldCounts().keySet().isEmpty() == false) {
+        if (results != null) {
```

Comment:
I had to change this so that at parsing time we can make the distinction between no existing results and empty results. 

When there is no results, the rendered output will be something like `"matrix_stats":{}` and empty results will be  `"matrix_stats":{"fields":[]}`. This allows us to reproduce the same behavior of MatrixStats methods in the transport client and parsed implementations.

I'm not sure that "no results" is a real use case but InternalMatrixStats has specific checks at several for `results` being null (and it's also randomized in tests) so it let me think that it can happen.